### PR TITLE
chore(proxy): clean up api crate

### PR DESCRIPTION
**Pull request by xla**
Assortment of cleanup commits in preparation of the great flippening.

### CODE REVIEW

**Review by rudolfs — APPROVED** [review]
🧹 💨

File: proxy/api/src/http.rs [hunk 1]
```diff
@@ -41,7 +40,6 @@ pub fn api(
 ) -> impl Filter<Extract = impl Reply, Error = Rejection> + Clone {
     let test = ctx.test();
 
-    let avatar_filter = path("avatars").and(avatar::get_filter());
```
Comment (rudolfs) [inline proxy/api/src/http.rs:44]: I think I'll need the `avatars` endpoint for https://github.com/radicle-dev/radicle-upstream/pull/1813 after all.

Comment (xla) [reply proxy/api/src/http.rs:44]: Generally we should reduce the dependence on the proxy going forward, esp. for non-essential features. If we need avatars for ETH addresses there should be readily available solutions employed in wallets like MetaMask.

Comment (FintanH) [reply proxy/api/src/http.rs:44]: In other words, `radicle-link`/`daemon` won't be maintaining an endpoint for avatars :)

Comment (rudolfs) [reply proxy/api/src/http.rs:44]: </3

Comment (FintanH) [reply proxy/api/src/http.rs:44]: :'(

Comment (xla) [reply proxy/api/src/http.rs:44]: :\

Comment (xla) [reply proxy/api/src/http.rs:44]: :/

Comment (xla) [reply proxy/api/src/http.rs:44]: :>

Comment (xla) [reply proxy/api/src/http.rs:44]: @rudolfs This concern is addressed since radicle-dev/radicle-avatar#5 landed, is that assumption correct?

Comment (rudolfs) [reply proxy/api/src/http.rs:44]: Yeah, we don't need the avatar endpoint anymore.

**Review by rudolfs — APPROVED** [review]

**Review by geigerzaehler — APPROVED** [review]

**Pull request merged**